Jenkinsfile

```diff
@@ -92,63 +144,117 @@ pipeline {
     }
     stage('Build and deploy preview') {
       when {
-        branch 'PR-*'
+        allOf {
+          branch 'PR-*'
+          not {
+            environment name: 'DRY_RUN', value: 'true'
+          }
+        }
       }
       steps {
         container('nodejs') {
-          script {
-            VERSION =  sh(script: 'npx -c \'echo "$npm_package_version"\'', returnStdout: true).trim()
-          }
-          withEnv(["VERSION=$VERSION-${BRANCH_NAME}"]) {
-            echo """
-              -----------------------------------
-              Building preview ${VERSION}
-              -----------------------------------"""
-            sh 'npx lerna run analysis --parallel'
-            dir('storybook') {
-              sh 'npx build-storybook -o dist -s ./public'
-              sh 'skaffold build'
-              dir('charts/preview') {
-                sh "make preview" // does some env subst before "jx step helm build"
-                sh "jx preview"
-              }
+          echo """
+            -----------------------------------
+            Building preview ${VERSION}
+            -----------------------------------"""
+          sh 'npx lerna run analysis --parallel'
+          dir('storybook') {
+            sh 'npx build-storybook -o dist -s ./public'
+            sh 'skaffold build'
+            dir('charts/preview') {
+              sh "make preview" // does some env subst before "jx step helm build"
+              sh "jx preview"
             }
           }
         }
       }
     }
-  }
-  post {
-    success {
-      container('nodejs') {
-        script {
-          if (BRANCH_NAME == 'master') {    
-             // publish elements
-             echo """
+    stage('Git commit, tag and push') {
+      when {
+        allOf {
+          not {
+            branch 'PR-*'
+          }
+          not {
+            environment name: 'DRY_RUN', value: 'true'
+          }
+        }
+      }
+      steps {
+        container('nodejs') {
+          echo """
+          --------
+          Git commit, tag and push
+          --------
+          """
+          sh """
+            #!/usr/bin/env bash -xe
```

Comment:
do we need to use bash and set `-xe`? is this not the default ?

Reply:
Not sure. Took it from here https://github.com/nuxeo/nuxeo/blob/master/ci/Jenkinsfiles/build.groovy#L644.